addons/website_event/models/event_event.py

```diff
@@ -290,49 +335,54 @@ def _update_website_menu_entry(self, fname_bool, fname_o2m, method_name):
 
         if self[fname_bool] and not self[fname_o2m]:
             # menus not found but boolean True: get menus to create
-            for sequence, menu_data in enumerate(getattr(self, method_name)()):
-                # some modules have 4 data: name, url, xml_id, menu_type; however we
-                # plan to support sequence in future modules, so this hackish code is
-                # necessary to avoid crashing. Not nice, but stable target = meh.
-                if len(menu_data) == 4:
-                    (name, url, xml_id, menu_type) = menu_data
-                    menu_sequence = sequence
-                elif len(menu_data) == 5:
-                    (name, url, xml_id, menu_sequence, menu_type) = menu_data
+            for menu_data in getattr(self, method_name)():
+                (name, url, xml_id, menu_sequence, menu_type) = menu_data
                 new_menu = self._create_menu(menu_sequence, name, url, xml_id, menu_type=menu_type)
         elif not self[fname_bool]:
             # will cascade delete to the website.event.menu
             self[fname_o2m].mapped('menu_id').sudo().unlink()
 
         return new_menu
 
-    def _create_menu(self, sequence, name, url, xml_id, menu_type=False):
-        """ If url: create a website menu. Menu leads directly to the URL that
-        should be a valid route. If xml_id: create a new page, take its url back
-        thanks to new_page of website, then link it to a menu. Template is
-        duplicated and linked to a new url, meaning each menu will have its own
-        copy of the template.
+    def _create_menu(self, sequence, name, url, xml_id, menu_type):
+        """ Create a new menu for the current event.
+
+        If url: create a website menu. Menu leads directly to the URL that
+        should be a valid route.
+
+        If xml_id: create a new page using the qweb template given by its
+        xml_id. Take its url back thanks to new_page of website, then link
+        it to a menu. Template is duplicated and linked to a new url, meaning
+        each menu will have its own copy of the template. This is currently
+        limited to two menus: introduction and location.
 
         :param menu_type: type of menu. Mainly used for inheritance purpose
-          allowing more fine-grain tuning of menus. """
+          allowing more fine-grain tuning of menus.
+        """
         self.check_access_rights('write')
+        view = self.env['ir.ui.view'].sudo()
```

Comment:
Don't we usually capitalize those? So "View"?
Also, seems a bit unnecessary to extract it as it's only used once.